documentation/rfc/ptref_grammar.md

```diff
@@ -0,0 +1,225 @@
+ 
+### Method predicates
+
+Method predicates are in the form `collection.method(string, string, ...)` where:
+ * `collection` is a collection name (as the collections queried, but singular);
+ * `method` is a method name, corresponding to the filter on the `collection`;
+ * arguments to the method between `(` and `)`, as strings, separated by `,`.
+ 
+Example: `/v1/coverage/id/stop_areas?filter=stop_area.has_code("neptune","1242")`
+ 
+#### Strings
+ 
+A string can be:
+ * an escaped string, begining and ending with `"`, with `\x` transformed as `x` within the string
+ * a basic string, composed of any letter, digit, `_`, `.`, `:`, `;`, `|` and `-`.
+ 
+Basic strings are great to easily type numbers and navitia coordinates. It can also handle most of the identifiers, but be warned that you will run into troubles if you use the basic string syntax in a programatic way (think of identifiers with spaces or accentuated letters).
+
+For programatic usage, it is recommended to:
+ * substitute `\` by `\\`
+ * substitute `"` by `\"` 
+ * surround the string with `"`
+
+Examples:
+ * `1337`
+ * `-42`
+ * `-123.45e-12`
+ * `2.37715;48.846781`
+ * `"I can have spaces and accentuated letters: éèà"` (decoded as `I can have spaces and accentuated letters: éèà`)
+ * `"a \" and a \\ within a string"` (decoded as `a " and a \ within a string`)
+
+#### Availlable methods
+
+In the following table, if the invocation starts with `collection`, any collection can be used, else the method will only work for the specified collection.
+
+| query | signification | note |
+|-------|---------------|------|
+|`collection.has_code(type, value)`|all the objects of type `collection` that have the code `{type: "type", value: "value"}` in `codes[]`|only the `codes[]` field is used|
+|`collection.id(id)`|the object of type `collection` that have `id` as identifier (empty if this identifier is not present)|for types without identifier (as `connection`), it is equivalent to `empty`|
+|`collection.uri(id)`|same as `collection.id(id)`|deprecated|
+|`collection.name(name)`|all the objects of type `collection` that have `name` as name|for types without name (as `connection`), it is equivalent to `empty`|
+|`disruption.between(since, until)`|all the disruptions that are active during the `[since, until]` period|`since` and `until` must be UTC datetime in the ISO 8601 format (ending with `Z`)|
+|`disruption.since(since)`|all the disruptions that are active after the given datetime|`since` must be UTC datetime in the ISO 8601 format (ending with `Z`)|
+|`disruption.tag(tag)`|all the disruptions containing the given tag|equivalent to `disruption.tags(tag)`|
+|`disruption.tags(tag1, tag2, ...)`|all the disruptions containing at least one of the given tags|at least one tag must be given|
+|`disruption.until(until)`|all the disruptions that are active before the given datetime|`until` must be UTC datetime in the ISO 8601 format (ending with `Z`)|
```

Comment:
better if the row is just after `between` and `since`

Reply:
That's in alphabetic order

Reply:
Not sure what's the most important, anyway it's fine :)

Reply:
reorganized